cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -356,15 +356,18 @@ func del(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 			log.Infof("Container %s not found", args.ContainerID)
 			return nil
 		}
-		log.Errorf("Error received from DelNetwork gRPC call for container %s: %v",
-			args.ContainerID, err)
+		log.Errorf("Error received from DelNetwork gRPC call for container %s: %v", args.ContainerID, err)
 
-		//  DelNetworkRequest may return a connection error, so try to delete using PrevResult whenever an error is returned.
+		// DelNetworkRequest may return a connection error, so try to delete using PrevResult whenever an error is returned. As with the case above, do
+		// not return error to kubelet, as there is no guarantee that delete is retried.
 		if teardownPodNetworkWithPrevResult(driverClient, conf, k8sArgs, args.IfName, log) {
```

Comment:
That sounds reasonable. I think it's best to file an issue and have that as a future refactor rather than doing in this PR, as this PR has gotten quite large.

Reply:
Filed https://github.com/aws/amazon-vpc-cni-k8s/issues/2374